### PR TITLE
updated oculus hand controller data provider

### PR DESCRIPTION
**Pull request by StephenHodgson**
Continued from https://github.com/XRTK/XRTK-Core/pull/476

### CODE REVIEW

**Comment by FejZa** [comment]
Can you elaborate those issues a bit? What kind of issues do you expect?

**Comment by SimonDarksideJ** [comment]
For one, we would have two sets of data providers polling the Oculus API, unclear whether that would have a direct impact.
But I've held off adding the hands nodes to the current provider. But when I did the copy approach, it causes loads of issues with unrecognised controllers (as there were no hands)

As the Oculus API is a singular dataset, my feeling is that the data provider updates should be incorporated in to the current provider, especially as only HANDS or Controllers can be active at the same time, which is managed automatically by the API at this time.

File: XRTK.Oculus/Packages/com.xrtk.oculus/Controllers/Hands/OculusHandControllerDataProvider.cs [hunk 1]
```diff
@@ -118,10 +114,10 @@ private MixedRealityHandController GetOrAddController(Handedness handedness)
                 return existingController;
             }
 
-            Type controllerType = typeof(MixedRealityHandController);
-            IMixedRealityPointer[] pointers = RequestPointers(controllerType, handedness, true);
-            IMixedRealityInputSource inputSource = MixedRealityToolkit.InputSystem.RequestNewGenericInputSource($"{handedness} Hand Controller", pointers);
-            MixedRealityHandController detectedController = new MixedRealityHandController(TrackingState.Tracked, handedness, inputSource, null);
+            var controllerType = typeof(MixedRealityHandController);
```
Comment (FejZa) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Controllers/Hands/OculusHandControllerDataProvider.cs]: Why put var everywhere needlessly?

**Comment by FejZa** [comment]
Exactly. The Oculus API manages the actual input sources. As soon as a controller is detected no more hand is reported to be avilable and vice versa. So no issues there. I can understand you want one data provider because they both share the "Oculus" in the name basically. I'll see if it's possible, the hand data providers do have some specifcs to them.

**Pull request merged**